libraries/Resample/Resample.cpp

```diff
@@ -0,0 +1,205 @@
+    else if (factor == 8 && quality == high && phase == minimum)
+        return fir_8_high_minphase;
+    else if (factor == 8 && quality == high && phase == linear)
+        return fir_8_high_linear;
+    else if (factor == 8 && quality == low && phase == minimum)
+        return fir_8_low_minphase;
+    else if (factor == 8 && quality == low && phase == linear)
+        return fir_8_low_linear;
+    else if (factor == 16 && quality == high && phase == minimum)
+        return fir_16_high_minphase;
+    else if (factor == 16 && quality == high && phase == linear)
+        return fir_16_high_linear;
+    else if (factor == 16 && quality == low && phase == minimum)
+        return fir_16_low_minphase;
+    else if (factor == 16 && quality == low && phase == linear)
+        return fir_16_low_linear;
+    else {
+        std::cerr << "get_fir: invalid combination " << factor << quality << phase << std::endl;
+        return std::vector<float>();
+    }
+}
+
+void ResampleBase::cleanup() {
+    NE10_FREE(pCoeff);
+    NE10_FREE(pState);
+    pCoeff = nullptr;
+    pState = nullptr;
+}
+
+int Decimator::setup(unsigned int decimationFactor, unsigned int blockSize, fir_quality quality, fir_phase phase) {
+    cleanup();
+    this->blockSizeIn = blockSize;
+    this->factor      = decimationFactor;
+    if (decimationFactor == 1) {
+        // Don't do anything
+        return 0;
+    }
+    std::vector<float> fir     = get_fir(decimationFactor, quality, phase);
+    uint               numTaps = fir.size();
+
+    pCoeff = (ne10_float32_t*)NE10_MALLOC(numTaps * sizeof(ne10_float32_t));
+    pState = (ne10_float32_t*)NE10_MALLOC(
+        (numTaps + blockSize - 1) * sizeof(ne10_float32_t));
+    if (!pState || !pCoeff)
+        return -1;
+
+    for (unsigned int n = 0; n < numTaps; ++n) {
+        // reverse filter coefficients
+        pCoeff[n] = fir[numTaps - 1 - n];
+    }
+    int err = ne10_fir_decimate_init_float(&decimator, numTaps, decimationFactor, pCoeff, pState, blockSize);
+    if (err) {
+        std::cerr << "ERR: blockSize must be whole multiple of decimationFactor" << std::endl;
+        return -1;
+    }
+
+    return 0;
+}
+
+void Decimator::process(ne10_float32_t* outBlock, const ne10_float32_t* inBlock) {
+    if (factor == 1 && outBlock != inBlock) {
+        memcpy(outBlock, inBlock, blockSizeIn * sizeof *outBlock);
+        return;
+    }
+    ne10_fir_decimate_float_neon(&decimator, (float*)inBlock, outBlock, blockSizeIn);
+}
+
+int Interpolator::setup(unsigned int L, unsigned int blockSize, fir_quality quality, fir_phase phase) {
+    cleanup();
+    this->blockSizeIn = blockSize / L;
+    this->factor      = L;
+    if (L == 1) {
+        // Don't do anything
+        return 0;
+    }
+    std::vector<float> fir      = get_fir(L, quality, phase);
+    uint               filtSize = fir.size();
+    // numTaps must be a multiple of the interpolation factor
+    uint numTaps     = ceil((float)filtSize / L) * L;
```

Comment:
not a fan of aligning the `=` becuase if you add a variable with a longer name you'll need to move everything else (or risk of leaving a mess as it happens at line 104 
```
    std::vector<float> fir      = get_fir(L, quality, phase);
```

Reply:
Got it.